### PR TITLE
Fix header links and sphinx warnings

### PR DESCRIPTION
**Pull request by meghanrjones**
**Description of proposed changes**

With the latest version of myst-parser, the header links in contributing.md no longer work. This PR follows the advice in https://github.com/executablebooks/MyST-Parser/issues/519 to fix the references.

Fixes #

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

**Review by seisman — DISMISSED** [review]

**Comment by seisman** [comment]
There are still two warnings:
```
/home/runner/work/pygmt/pygmt/doc/contributing.md:424: WARNING: hardcoded link '[docs.generic-mapping-tools.org/latest/makecpt.html](https://docs.generic-mapping-tools.org/latest/makecpt.html)' could be replaced by an extlink (try using ':gmt-docs:`makecpt.html`' instead)
/home/runner/work/pygmt/pygmt/doc/contributing.md:424: WARNING: hardcoded link '[docs.generic-mapping-tools.org/latest/gmt.conf.html#term-COLOR_FOREGROUND](https://docs.generic-mapping-tools.org/latest/gmt.conf.html#term-COLOR_FOREGROUND)' could be replaced by an extlink (try using ':gmt-docs:`gmt.conf.html#term-COLOR_FOREGROUND`' instead)
```

**Comment by meghanrjones** [comment]
> There are still two warnings:
> 
> ```
> /home/runner/work/pygmt/pygmt/doc/contributing.md:424: WARNING: hardcoded link '[docs.generic-mapping-tools.org/latest/makecpt.html](https://docs.generic-mapping-tools.org/latest/makecpt.html)' could be replaced by an extlink (try using ':gmt-docs:`makecpt.html`' instead)
> /home/runner/work/pygmt/pygmt/doc/contributing.md:424: WARNING: hardcoded link '[docs.generic-mapping-tools.org/latest/gmt.conf.html#term-COLOR_FOREGROUND](https://docs.generic-mapping-tools.org/latest/gmt.conf.html#term-COLOR_FOREGROUND)' could be replaced by an extlink (try using ':gmt-docs:`gmt.conf.html#term-COLOR_FOREGROUND`' instead)
> ```

Thanks, this is now fixed.

File: doc/contributing.md [hunk 1]
```diff
@@ -97,7 +97,7 @@ where you can submit general comments and/or questions:
 ### Resources for New Contributors
```
Comment (seisman) [inline doc/contributing.md:97]: Please fix a typo (`Sow` => `Show`) at line 93. 
Changing the URL to `https://forum.generic-mapping-tools.org/c/Show-your-nice-example-script/10`.

**Review by seisman — APPROVED** [review]
Looks good.

**Pull request merged**